src/util/obbt.jl

```diff
@@ -0,0 +1,406 @@
+
+        # bound-tightening for the td variables
+        for bp in buspairs
+            (td_ub[bp] - td_lb[bp] < min_bound_width) && (continue)
+
+            # td lower bound solve 
+            lb = NaN 
+            @objective(model_bt.model, Min, td[bp])
+            result_bt = solve_generic_model(model_bt, solver)
+            if (result_bt["status"] == :LocalOptimal || result_bt["status"] == :Optimal)
+                nlb = floor(output_tol * getobjectivevalue(model_bt.model))/output_tol
+                (nlb > td_lb[bp]) && (lb = nlb)
+            end
+
+            # td upper bound solve 
+            ub = NaN 
+            @objective(model_bt.model, Max, td[bp])
+            result_bt = solve_generic_model(model_bt, solver)
+            if (result_bt["status"] == :LocalOptimal || result_bt["status"] == :Optimal)
+                nub = ceil(output_tol * getobjectivevalue(model_bt.model))/output_tol
+                (nub < td_ub[bp]) && (ub = nub)
+            end 
+
+            # sanity checks 
+            (lb > ub) && (warn(LOGGER, "bt lb > ub - adjust tolerances in solver to avoid issue"); continue)
+            (!isnan(lb) && lb > td_ub[bp]) && (lb = td_lb[bp])
+            (!isnan(ub) && ub < td_lb[bp]) && (ub = td_ub[bp])
+            isnan(lb) && (lb = td_lb[bp])
+            isnan(ub) && (ub = td_ub[bp])
+
+            # td bound-reduction computation
+            td_reduction = 0.0
+            if (ub - lb >= min_bound_width)
+                td_reduction = (td_ub[bp] - td_lb[bp]) - (ub - lb)
+                td_lb[bp] = lb 
+                td_ub[bp] = ub 
+            elseif (td_ub[bp] - lb >= min_bound_width)
+                td_reduction = (td_ub[bp] - td_lb[bp]) - (td_ub[bp] - lb)
+                td_lb[bp] = lb 
+            elseif (ub - td_lb[bp] >= min_bound_width) 
+                td_reduction = (td_ub[bp] - td_lb[bp]) - (ub - td_lb[bp])
+                td_ub[bp] = ub
+            else 
+                td_reduction = 0.0
+            end 
+
+            # sequential obbt model update
+            if td_reduction != 0.0 && sequential_obbt
+                modifications = create_modifications(model_bt, td_lb[bp], td_ub[bp], bp)
+                PowerModels.update_data(data, modifications)
+                model_bt = build_generic_model(data, model_constructor, PowerModels.post_opf)
+                (upper_bound_constraint) && (constraint_obj_bound(model_bt, upper_bound))
+                vm = var(model_bt, :vm)
+                td = var(model_bt, :td)
+            end
+            total_td_reduction += (td_reduction)
+            max_td_reduction = max(td_reduction, max_td_reduction)
+
+        end 
+        avg_td_reduction = total_td_reduction/length(buspairs)
+
+        td_range_final = sum([td_ub[bp] - td_lb[bp] for bp in buspairs])
+        
+        info(LOGGER, "iteration $(current_iteration+1), vm range: $vm_range_final, td range: $td_range_final")
```

Comment:
Would the objective value of the current relaxation may also be of interest?

Reply:
agreed, logging the objective value in the updated PR